### DISCUSSION TITLE
Implement contraints on weights of the NN (something similar to minmax norm from Keras)

### DISCUSSION THREAD

**Discussion by NischalKMapakshi** [lululxvi/deepxde#1260] [General]
Dear Lulu,

I'm trying to solve Darcy equations imposing maximum principles on pressure. In a nutshell I have a square domain with circular hole. Dirichlet BC on both inner and outer boundaries. My solutions for pressure should be pmin<=p<=pmax. Where pmin and pmax will be on the boundaries. While using CNN from Keras we implemented this by adding an additional dense layer to the NN and placing constraint on the weights for that layer (basically minmax norm from tensorflow). I'm not able to do this using DeepXDE.

Please let me know how this can be achieved using deepxde or is there a better way of doing this. 
PS: 
Placing constraints on the weights for all the layers can also help (I think so). Please direct me to any example if available.

Thank you,

Nischal

**Comment by lululxvi** [comment]
There are different ways to do it. If you already know how to do this in Keras, then you can modify the code of DeepXDE in the tensorflow backend. Basically, you can add another loss.
